.github/workflows/publish.yaml

```diff
@@ -0,0 +1,89 @@
+name: Publish Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: 'Version of the latest release (e.g. v0.42.0)'
+        required: false
+        default: ''
+  repository_dispatch:
+    types: [ocm-cli-release]
+
+jobs:
+
+  push-to-aur:
+    name: Update Arch Linux User Repository
+    runs-on: ubuntu-latest
+    steps:
+    - name: Ensure proper version
+      run: |
+        if [ -z "${{ github.event.client_payload.version }}" ]; then
+          if [ -z "${{ github.event.inputs.version }}" ]; then
+            echo "Version not provided"
+            exit 1
+          fi
+          echo "RELEASE_VERSION=$(echo ${{ github.event.inputs.version }} | tr -d ['v'])" >> $GITHUB_ENV
+        else
+          echo "RELEASE_VERSION=$(echo ${{ github.event.client_payload.version }} | tr -d ['v'])" >> $GITHUB_ENV
+        fi
+        cat $GITHUB_ENV
+    - name: Install SSH key
+      uses: shimataro/ssh-key-action@v2
+      with:
+        name: aur
+        key: ${{ secrets.ARCHLINUX_SSH_PRIVATE_KEY }}
+        known_hosts: ${{ secrets.ARCHLINUX_SSH_KNOWN_HOSTS }}
+        config: ${{ secrets.ARCHLINUX_SSH_CONFIG}}
+    - name: Setup git config
+      run: |
+        git config user.name "GitHub Actions Bot"
+        git config user.email "<41898282+github-actions[bot]@users.noreply.github.com>"
+    - name: Git operations with SSH
+      run: |
+        git clone ssh://aur@aur.archlinux.org/ocm-cli.git
+        cd ocm-cli
+        ./upgrade ${{ env.RELEASE_VERSION }}
+        git add PKGBUILD .SRCINFO
+        git commit -m"Release ${{ env.RELEASE_VERSION }}"
+        git push origin @:refs/heads/master
+
+  push-to-chocolatey:
+    name: Update Chocolatey
+    runs-on: windows-latest
+    steps:
+    - name: Ensure proper version
+      run: |
+        $workflow_version = "${{ github.event.inputs.version }}"
```

Comment:
Why does this step `Ensure proper version` differ from the step above? Is this necessary? Could we use a reusable workflow here?

Reply:
because it's within a different job, which runs on a different runner - nope, I don't think there is a reusable way... unless you have an idea?